### PR TITLE
feat: integrate viper with cli

### PR DESCRIPTION
**Pull request by zoumo**
**What this PR does**
integrate viper with cli

**why we need it**

**Special notes for your reviewer**:

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -40,3 +40,7 @@
 [[constraint]]
   name = "github.com/spf13/viper"
   version = "1.0.0"
+
+# [[override]]
```
Comment (Jimexist) [inline Gopkg.toml]: remove unused code

File: Gopkg.lock [hunk 2]
```diff
@@ -1,6 +1,12 @@
 # This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.
 
 
+[[projects]]
+  name = "github.com/davecgh/go-spew"
```
Comment (ddysher) [inline Gopkg.lock]: `go-spew` and `go-difflib` is for debugging?  They don't seem to be used below.

File: cli/flag.go [hunk 3]
```diff
@@ -17,16 +17,13 @@ limitations under the License.
 package cli
 
 import (
+	"fmt"
```
Comment (ddysher) [inline cli/flag.go]: Changes below has nothing to do with viper integration?  Please create a separate PR.

File: cli/mamba.go [hunk 4]
```diff
@@ -18,12 +18,6 @@ package cli
 
 import (
 	"github.com/spf13/cobra"
-	"github.com/spf13/viper"
-)
-
```
Comment (ddysher) [inline cli/mamba.go]: Don't use `mamba` (I like this name though)

File: cli/util.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+func stringInSlice(a string, list []string) bool {
```
Comment (ddysher) [inline cli/util.go]: Add comment

File: cli/viper.go [hunk 6]
```diff
@@ -0,0 +1,133 @@
+/*
```
Comment (ddysher) [inline cli/viper.go]: Don't use file name `viper`.

File: examples/cli/main.go [hunk 7]
```diff
@@ -32,30 +37,26 @@ func main() {
 		Use:  "example",
 		Long: "this is an cli example",
 		Run: func(cmd *cobra.Command, args []string) {
-			flog := cmd.PersistentFlags().Lookup("log")
+			flog := cmd.LocalFlags().Lookup("log")
```
Comment (ddysher) [inline examples/cli/main.go]: Is this change necessary for integration?

**Comment by ddysher** [comment]
@Jimexist  

 coveralls 👈   he is very noisy.

File: Gopkg.lock (hunk 2 above)
Comment (zoumo) [reply Gopkg.lock]: they are used by assert

File: examples/cli/main.go (hunk 7 above)
Comment (ddysher) [reply examples/cli/main.go]: Ping?
